### PR TITLE
neovim: update to 0.10.0

### PR DESCRIPTION
**Pull request by tranzystorekk**
#### Testing the changes
- I tested the changes in this PR: YES

Neovim now depends on lua-lpeg as a library and dynamically links to it, but we don't serve it as a shlib.

### CODE REVIEW

**Comment by classabbyamp** [comment]
workaround:
```diff
- riscv64*) build_options_default="" ;;
+ armv7l|riscv64*) build_options_default="" ;;
```

**Comment by AnInternetTroll** [comment]
Tested on x86\_64-glibc with some LSPs. Works as expected, though I had
to :TSInstall some parsers again. Anyway, great work!

**Comment by classabbyamp** [comment]
> though I had to :TSInstall some parsers again

probably more related to the ABI break in treesitter

**Comment by ibhagwan** [comment]
@tranzystorekk, ty for your work on this.

This build however is incomplete, it is missing lua the default treesitter parsers:
```
/lib/nvim/parser/bash.so
/lib/nvim/parser/c.so
/lib/nvim/parser/lua.so
/lib/nvim/parser/markdown.so
/lib/nvim/parser/markdown_inline.so
/lib/nvim/parser/python.so
/lib/nvim/parser/query.so
/lib/nvim/parser/vim.so
/lib/nvim/parser/vimdoc.so
```

This can result in https://github.com/neovim/neovim/issues/27951,  which you may not experience if you have `nvim-treesitter` plugin installed as it will compile the missing parsers.

**Comment by tranzystorekk** [comment]
That would require packaging these parsers separately, and I don't have the bandwidth for that.

**Comment by ibhagwan** [comment]
> That would require packaging these parsers separately, and I don't have the bandwidth for that.

Why separately? When I build neovim locally and these are built inside the neovim build folder, isn’t this as simple as adding these files to the package?

Without these files the package is incomplete and will generate an error for example when opening lua files:
IMG_2104

**Comment by tranzystorekk** [comment]
> Why separately? When I build neovim locally and these are built inside the neovim build folder, isn’t this as simple as adding these files to the package?

we specifically don't invoke the `cmake.deps` step, and furthermore bundled deps need to be downloaded and our build_style cmake **specifically** disables network access so that doesn't happen.

**Comment by ibhagwan** [comment]
> > Why separately? When I build neovim locally and these are built inside the neovim build folder, isn’t this as simple as adding these files to the package?
> 
> we explicitly don't invoke the `cmake.deps` step, and furthermore bundled deps need to be downloaded and our build_style cmake **specifically** disables network access so that doesn't happen.

So do you think this can be released as is without the libs or should we block merging of this package for a while?

Having an error and no highlights when opening a lua/markdown/etc file isn’t really acceptable IMHO.

**Comment by tranzystorekk** [comment]
> So do you think this can be released as is without the libs or should we block merging of this package for a while?

Now that I agree shouldn't happen.
I'll experiment with manually downloading the needed parser distfiles and invoking additional build steps to get them into neovim.

**Comment by ibhagwan** [comment]
Ty @tranzystorekk for the latest pushes, work perfectly on glibc x86_64.

File: srcpkgs/neovim/template [hunk 1]
```diff
@@ -35,13 +70,27 @@ alternatives="
 # They want assertion
 CFLAGS=-UNDEBUG
 
+post_extract() {
+	local _lang
+	for _lang in c lua vim vimdoc query python bash markdown; do
+		local _varname="_treesitter_${_lang}_version"
+		local _version="${!_varname}"
+		vsrcextract -C .deps/build/src/treesitter_${_lang} ts_${_lang}_${_version}.tar.gz
+	done
```
Comment (classabbyamp) [inline srcpkgs/neovim/template:79]:
```suggestion
	for _lang in $skip_extraction; do
		vsrcextract -C ".deps/build/src/${_lang##*_}" "${_lang}"
	done
```
maybe something like this? (and i believe for loops make implicit locals)

**Comment by vazw** [comment]
I got these error while installing

```sh
-- Could NOT find Lua (missing: LUA_LIBRARIES LUA_INCLUDE_DIR) (Required is exact version "5.1")
CMake Error at /usr/libexec/xbps-src/share/cmake-3.27/Modules/ExternalProject.cmake:3176 (message):
  No download info given for 'treesitter_c' and its source directory:

   /builddir/neovim-0.10.0/.deps/build/src/treesitter_c

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /usr/libexec/xbps-src/share/cmake-3.27/Modules/ExternalProject.cmake:4345 (_ep_add_download_command)
  cmake/BuildTreesitterParsers.cmake:21 (ExternalProject_Add)
  cmake/BuildTreesitterParsers.cmake:31 (BuildTSParser)
  CMakeLists.txt:155 (include)


-- Configuring incomplete, errors occurred!
=> ERROR: neovim-0.10.0_1: pre_configure: 'cmake -S cmake.deps -B .deps -G Ninja -DUSE_BUNDLED=OFF -DUSE_BUNDLED_TS_PARSERS=ON' exited with 1
=> ERROR:   in pre_configure() at srcpkgs/neovim/template:81
```

**Pull request merged**